bundles/init/src/main/java/com/adobe/cq/commerce/demandware/init/InitServlet.java

```diff
@@ -101,20 +104,21 @@ protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse
             request.getRequestPathInfo().getExtension());
     }
 
-    private void upload(SlingHttpServletRequest slingRequest, String endPoint, String assetURI) {
+    private void upload(SlingHttpServletRequest slingRequest, String endPoint, String assetURI,
+                        DemandwareClient dwClient) {
         final RequestBuilder requestBuilder = RequestBuilder.get();
         requestBuilder.setUri(assetURI);
         final Cookie token = slingRequest.getCookie("login-token");
         requestBuilder.addHeader(new BasicHeader("Cookie", token.getName() + "=" + token.getValue()));
         final HttpHost localHost = new HttpHost("localhost", 4502);
-        final HttpClient httpClient = demandwareClient.getHttpClientBuilder().build();
+        final HttpClient httpClient = clientProvider.getDefaultClient().getHttpClientBuilder().build();
         try {
             // get the from AEM content
             final HttpResponse response = httpClient.execute(localHost, requestBuilder.build());
 
             if (response != null && response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
                 final HttpEntity entity = response.getEntity();
-                HttpClientBuilder httpClientBuilder = demandwareClient.getHttpClientBuilder();
+                HttpClientBuilder httpClientBuilder = dwClient.getHttpClientBuilder();
```

Comment:
why are we using "dwClient" here and the "defaultClient" above?

Reply:
hi, I fixed this in dev-with-preview branch, as soon as we merge it to dev it should be ok.